test/integration/operator_test.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/kubearchive/kubearchive/test"
+	corev1 "k8s.io/api/core/v1"
+	errs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Test int
+
+const (
+	emptyKAC Test = iota
+	nonEmptyKAC
+)
+
+const (
+	kacName     = "kubearchive"
+	a13eName    = kacName + "-a13e"
+	sinkName    = kacName + "-sink"
+	filtersName = "sink-filters"
+)
+
+func TestEmptyKAC(t *testing.T) {
+	runTest(t, emptyKAC, map[string]any{})
+}
+
+func TestNonEmptyKAC(t *testing.T) {
+	resources := map[string]any{
+		"resources": []map[string]any{
+			{
+				"selector": map[string]string{
+					"apiVersion": "v1",
+					"kind":       "Pod",
+				},
+				"archiveWhen": "true",
+				"deleteWhen":  "status.phase == 'Succeeded'",
+			},
+		},
+	}
+	runTest(t, nonEmptyKAC, resources)
+}
+
+func runTest(t *testing.T, testName Test, resources map[string]any) {
+	clientset, dynaclient, errClient := test.GetKubernetesClient()
+	if errClient != nil {
+		t.Fatal(errClient)
+	}
+
+	// Create test namespace
+	namespaceName := fmt.Sprintf("test-%s", test.RandomString())
+	t.Log("Running test in namespace "+namespaceName)
+	_, errNamespace := clientset.CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}, metav1.CreateOptions{})
+	if errNamespace != nil {
+		t.Fatal(errNamespace)
+	}
+	// This register the function.
+	t.Cleanup(func() {
+		// delete the test namespace
+		errNamespace = clientset.CoreV1().Namespaces().Delete(context.Background(), namespaceName, metav1.DeleteOptions{})
+		if errNamespace != nil {
+			t.Fatal(errNamespace)
+		}
+
+		retryErr := retry.Do(func() error {
+			_, getErr := clientset.CoreV1().Namespaces().Get(context.Background(), namespaceName, metav1.GetOptions{})
+			if !errs.IsNotFound(getErr) {
+				return errors.New("Waiting for namespace "+namespaceName+" to be deleted")
+			}
+			return nil
+		}, retry.Attempts(10), retry.MaxDelay(3*time.Second))
+
+		if retryErr != nil {
+			t.Log(retryErr)
+		}
+	})
+
+	kac := createKAC(namespaceName, resources)
+	gvr := kubearchivev1alpha1.GroupVersion.WithResource("kubearchiveconfigs")
+	_, err := dynaclient.Resource(gvr).Namespace(namespaceName).Create(context.Background(), kac, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = checkResourcesAfterApply(namespaceName, testName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = dynaclient.Resource(gvr).Namespace(namespaceName).Delete(context.Background(), kacName, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = checkResourcesAfterDelete(namespaceName, testName)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func checkResourcesAfterApply(namespace string, testName Test) error {
+	clientset, dynaclient, _ := test.GetKubernetesClient()
```

Comment:
Since `checkResourcesAfterApply` is essentially a helper function for `runTest`, I think it could be made simpler by doing the following:

```suggestion
func checkResourcesAfterApply(t testing.TB, namespace string, testName Test) {
	t.Helper()
	
	clientset, dynaclient, _ := test.GetKubernetesClient()
```

because `t.Helper()` is called, instead of having `checkResourcesAfterApply` return errors it encounters, it can call `t.Fatal` or `assert.<whatever>` itself instead of the `runTests` having to check a return value